sound/soc/sof/ipc4-topology.h

```diff
@@ -107,6 +107,11 @@ enum sof_ipc4_copier_module_config_params {
 	SOF_IPC4_COPIER_MODULE_CFG_ATTENUATION,
 };
 
+struct sof_ipc4_prepare_copier_tuples {
+	u32 deep_buffer_dma_ms;
+	bool fast_mode;
+};
```

Comment:
is it certain that the fast mode is only usable with deep buffer? Based on my tests on the reference IPC4 platform, fast mode is selectable for non deepbuffer PCMs.

Reply:
The "ipc4_copier_deep_buffer_tokens"-naming is unfortunate. The token list is in no way tied to deep buffer pipeline copiers. I have little trouble in understanding how the token lists should be used. They are planned for a simple case where object tuples can be used to fill up a simple msg struct fields, but is not often the case with IPC4 messaging. Anyway, maybe I should rename SOF_COPIER_TOKENS and ipc4_copier_tokens to SOF_COPIER_NODE_TYPE_TOKENS and ipc4_copier_node_type_tokens, and make SOF_COPIER_DEEP_BUFFER_TOKENS simply SOF_COPIER_TOKENS... hmm or actually make SOF_COPIER_TOKENS SOF_COPIER_SETUP_PCM_TOKENS, and SOF_COPIER_DEEP_BUFFER_TOKENS SOF_COPIER_PREPARE_TOKENS.

Reply:
It is still not clear if this only applies to host copier or can be set for the buffer types also.
If only host copier can have this then have it named as SOF_HOST_COPIER_TOKENS?

Reply:
The tokens are parsed for all tplg_ipc4_widget_ops[] table entries that have SOF_COPIER_DEEP_BUFFER_TOKENS in their token_list[], and the values are extracted if the related functions call sof_update_ipc_object(). The system is a bit hairy, but I think I understand the Linux side now (I've been reverse engineering the code more than a month now), the topo2 side still manages to surprise me from time to time  😅.